### PR TITLE
Add/remove remote tracks from msid streams based on direction (rebased)

### PR DESCRIPTION
**Pull request by jan-ivar**
Rebase of https://github.com/w3c/webrtc-pc/pull/1377.
Fix for https://github.com/w3c/webrtc-pc/issues/1181.

***
[Preview](https://s3.amazonaws.com/pr-preview/jan-ivar/webrtc-pc/removeremote2.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/webrtc-pc/a12bf47...jan-ivar:f13a05e.html)

### CODE REVIEW

**Comment by jan-ivar** [comment]
@taylor-b Please double-check the last commit. Thanks.

**Comment by jan-ivar** [comment]
@stefhak @taylor-b Can we merge this? All feedback from https://github.com/w3c/webrtc-pc/pull/1377 has been incorporated.

File: webrtc.html [hunk 1]
```diff
@@ -1586,22 +1586,20 @@
                         "applying-a-remote-desc">[[!JSEP]]</span>.</p>
                       </li>
                       <li>
-                        <p>If the direction of the media description as defined
-                        in
-                        <span data-jsep="applying-a-remote-desc">[[!JSEP]]</span>
-                        is <code>sendrecv</code> or <code>sendonly</code>, and
+                        <p>If <var>direction</var> is <code>sendrecv</code> or
+                        <code>recvonly</code>, and
```
Comment (taylor-b) [inline webrtc.html]: Should be "sendonly"

File: webrtc.html [hunk 2]
```diff
@@ -1586,22 +1586,20 @@
...
                         <var>transceiver</var>'s
                         [[<a>CurrentDirection</a>]] slot
-                        is neither "sendrecv" nor "sendonly",
+                        is neither "sendrecv" nor "recvonly",
```
Comment (taylor-b) [inline webrtc.html]: This is fine though (remember, "direction" above is the reversed direction from a remote description, whereas CurrentDirection isn't... I wonder if we could do something to make this less confusing).

File: webrtc.html [hunk 3]
```diff
@@ -1586,22 +1586,20 @@
...
                         <a>process the addition of a remote track</a> for
                         the media description, given <var>transceiver</var>.</p>
                       </li>
                       <li>
-                        <p>If the direction of the media description is
-                        <code>recvonly</code> or <code>inactive</code>, and
+                        <p>If <var>direction</var> is
+                        <code>sendonly</code> or <code>inactive</code>, and
```
Comment (taylor-b) [inline webrtc.html]: Should be "recvonly"

File: webrtc.html [hunk 4]
```diff
@@ -5558,6 +5576,72 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
...
+            <code><a>MediaStream</a></code> objects created for this
+            <var>connection</var> with the <code>id</code>s corresponding to
+            <var>msids</var>.</p>
+          </li>
+          <li>
+            <p>For each <var>stream</var> in <var>receiver</var>'s
+            [[<a>associated remote MediaStreams</a>]] that is not present in
+            <var>streams</var>, remove <var>track</var> from <var>stream</var>.
+            <p class="note">This will result in a removetrack event being fired
+            at each MediaStream as described in [[!GETUSERMEDIA]].</p>
```
Comment (taylor-b) [inline webrtc.html]: I think there's a missing `</p>` here.

File: webrtc.html (hunk 1 above)
Comment (jan-ivar) [inline webrtc.html]: @taylor-b I reversed it to match github tip's [**set an RTCSessionDescription**](https://rawgit.com/w3c/webrtc-pc/master/webrtc.html#set-description)'s step 2.2.8.1., which says:
 * *"Let* direction *be an RTCRtpTransceiverDirection value representing the direction from the media description, but with the send and receive directions reversed to represent this peer's point of view."*

So unless I'm royally confused, wouldn't a remote offer with incoming media have "sendonly" and cause *direction* to be *recvonly* here?

File: webrtc.html (hunk 3 above)
Comment (jan-ivar) [inline webrtc.html]: See above.

File: webrtc.html (hunk 1 above)
Comment (taylor-b) [inline webrtc.html]: Sorry, I missed the direction reversal step somehow.

File: webrtc.html [hunk 5]
```diff
@@ -6695,6 +6780,13 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
           <p>Set <var>receiver.track</var> to <var>track</var>.</p>
         </li>
         <li>
+          <p>Let <var>receiver</var> have an [[<dfn>associated remote
+          MediaStreams</dfn>]] internal slot, representing a list of
```
Comment (taylor-b) [inline webrtc.html]: Should use new internal slot naming scheme.

**Comment by taylor-b** [comment]
Can you fix the conflicts? Then I think we'll be ready to merge this.

**Comment by jan-ivar** [comment]
@taylor-b Great! I tried to click the `Resolve conflicts` button here before, but I found no way to resolve things without pushing at the same time. I can try rebase again I suppose, just a bit scared.

**Comment by jan-ivar** [comment]
@taylor-b Rebased.

**Comment by jan-ivar** [comment]
@taylor-b Good to go?

**Comment by taylor-b** [comment]
@jan-ivar Yep, looks good to me.

**Comment by jan-ivar** [comment]
@aboba Anything else needed here? I'm away next week.

**Comment by stefhak** [comment]
FWIW, LGTM.

**Pull request merged**